tests/unit/commands/test_names.py

```diff
@@ -199,22 +200,83 @@ def test_register_store_id(emitter, fake_store_register):
     ],
 )
 @pytest.mark.usefixtures("memory_keyring")
-def test_names(emitter, fake_store_get_names, command_class):
-    cmd = command_class(None)
+class TestNames:
+    """Tests for the names command."""
+
+    def test_table(self, emitter, fake_store_get_names, command_class):
+        cmd = command_class(None)
+
+        cmd.run(
+            argparse.Namespace(
+                store_id="1234",
+                private=False,
+                yes=True,
+                format="table",
+                **{"snap-name": "test-snap"},
+            )
+        )
 
-    cmd.run(
-        argparse.Namespace(
-            store_id="1234", private=False, yes=True, **{"snap-name": "test-snap"}
+        if command_class.hidden:
+            emitter.assert_progress("This command is deprecated: use 'names' instead")
+        emitter.assert_message(
+            dedent(
+                """\
+                Name               Since                 Visibility    Notes
+                test-snap-private  2016-07-26T20:18:32Z  private       -
+                test-snap-public   2016-07-26T20:18:32Z  public        -"""
+            )
         )
-    )
 
-    if command_class.hidden:
-        emitter.assert_progress("This command is deprecated: use 'names' instead")
-    emitter.assert_message(
-        dedent(
-            """\
-            Name               Since                 Visibility    Notes
-            test-snap-private  2016-07-26T20:18:32Z  private       -
-            test-snap-public   2016-07-26T20:18:32Z  public        -"""
+    def test_json(self, emitter, fake_store_get_names, command_class):
+        cmd = command_class(None)
+
+        cmd.run(
+            argparse.Namespace(
+                store_id="1234",
+                private=False,
+                yes=True,
+                format="json",
+                **{"snap-name": "test-snap"},
+            )
         )
-    )
+
+        if command_class.hidden:
+            emitter.assert_progress("This command is deprecated: use 'names' instead")
+
+        emitter.assert_message(
+            json.dumps(
+                {
+                    "snaps": [
+                        {
+                            "name": "test-snap-private",
+                            "since": "2016-07-26T20:18:32Z",
+                            "visibility": "private",
+                            "notes": "-",
```

Comment:
if notes is `-`, from a data model perspective, should this not just be nil?

Reply:
I thought it strange that the data produced by `snaps = store_client.get_names()` set `notes` to `-`. I expected it to contain a `None`/`nil`/`null` value and `tabulate` to render that to `-`.  I decided to not transform the field so as to remain consistent with both the store and the table format. 

Should it be `null`? Yes I think it probably should. If there's appetite for this, I'll update the PR.

Reply:
Yeah, do you mind making this change for the json output?  The table can remain as-is with the "-".

I think it should be an easy change, something along the lines of:
```diff
--- a/snapcraft/commands/names.py
+++ b/snapcraft/commands/names.py
@@ -59,6 +59,10 @@ _MESSAGE_REGISTER_SUCCESS = "Registered {!r}"
 _MESSAGE_REGISTER_NO = "Snap name {!r} not registered"


+def _set_nil(value):
+    return None if value == "-" else value
+
+
 class StoreRegisterCommand(AppCommand):
     """Register a snap name with the Snap Store."""

@@ -161,7 +165,7 @@ class StoreNamesCommand(AppCommand):
             elif parsed_args.format == "json":
                 json_snaps = {
                     "snaps": [
-                        {header.lower(): snap[n] for n, header in enumerate(headers)}
+                        {header.lower(): _set_nil(snap[n]) for n, header in enumerate(headers)}
                         for snap in snaps
                     ]
```

Reply:
@mr-cal 

> do you mind making this change for the json output?

I'll happily update it in the next couple of days :)

Last time I force pushed to my own branch and that propagated through to the PR. Would it be sufficient to do that again?

Reply:
Yep, that should do the trick.  Thanks!

Reply:
@mr-cal I've updated the original commit and force pushed it.

Reply:
Thank you!